### PR TITLE
[WIP] Chunk encryption

### PR DESCRIPTION
**Pull request by gbalint**
DO NOT MERGE PR opened, code in early phase

### CODE REVIEW

File: swarm/storage/encryption/encryption_test.go [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+var hashFunc = sha3.NewKeccak256
+
+func TestEncryptDataLongerThanPadding(t *testing.T) {
+	enc := New(4095, uint32(0), hashFunc)
+
+	data := make([]byte, 4096)
+	key := make([]byte, 32)
+
+	_, err := enc.Encrypt(data, key)
+	if err == nil || !strings.Contains(err.Error(), "Data length longer than padding") {
```
Comment (zelig) [inline swarm/storage/encryption/encryption_test.go]: why use contains and not `err.Error() != expMsg`

File: swarm/storage/encryption/encryption_test.go [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+	data := make([]byte, 4097)
+	key := make([]byte, 32)
+
+	_, err := enc.Decrypt(data, key)
+	if err == nil || !strings.Contains(err.Error(), "Data length different than padding") {
+		t.Fatalf("Expected error \"Data length different than padding\" got %v", err)
+	}
+}
+
+func TestEncryptDecryptIsIdentity(t *testing.T) {
```
Comment (zelig) [inline swarm/storage/encryption/encryption_test.go]: use `rand.Read` to test many. and create benchmark for both ways.

**Review by zelig — CHANGES_REQUESTED** [review]

File: swarm/storage/encryption/encryption_test.go (hunk 1 above)
Comment (gbalint) [reply swarm/storage/encryption/encryption_test.go]: Fixed in f03d355cd48648436487c8c9d7c3343c196c2e9f

File: swarm/storage/encryption/encryption_test.go (hunk 2 above)
Comment (gbalint) [reply swarm/storage/encryption/encryption_test.go]: Sure, these are not all the tests I wanted to write, just the beginning

**Pull request closed without merging**